editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -0,0 +1,137 @@
+        if (project != null) {
+        	IJavaProject jproject = JavaCore.create(project);
+	        if (jproject == null) {
+	            scope = SearchEngine.createWorkspaceScope();
+	        } else {
+                scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { jproject });
+            }
+        }
+
+        try {
+            SelectionDialog dialog = JavaUI.createTypeDialog(shell, null, scope,
+                    IJavaElementSearchConstants.CONSIDER_CLASSES, false, "*Bean"); //$NON-NLS-1$
+            if (dialog.open() == SelectionDialog.OK) {
+                Object[] result = dialog.getResult();
+                if (result.length > 0 && result[0] instanceof IType) {
+                    return ((IType) result[0]).getFullyQualifiedName();
+                }
+            }
+        } catch (JavaModelException e) {
+			CamelEditorUIActivator.pluginLog().logError(e);
+        }
+        return null;
+    }
+
+	private static String openMethodDialog(IJavaProject jproject, String className, Shell shell) throws JavaModelException {
+		IType foundClass = jproject.findType(className);
+		if (foundClass != null) {
+			IMethod[] methods = foundClass.getMethods();
+			ElementListSelectionDialog dialog = new ElementListSelectionDialog(shell, new JavaUILabelProvider());
+			dialog.setTitle("Method Selection");
+			dialog.setMessage("Select a Method:");
+			dialog.setElements(methods);
+			if (dialog.open()  == SelectionDialog.OK) {
+				IMethod result = (IMethod) dialog.getFirstResult();
+				return result.getElementName();
+			}
+		}
+		return null;
+	}
+	
+    public static String handleMethodBrowse(IProject project, String className, Shell shell) {
+        if (project != null) {
+        	IJavaProject jproject = JavaCore.create(project);
+	        if (jproject.exists()) {
+		        try {
+		        	return openMethodDialog(jproject, className, shell);
+				} catch (JavaModelException e) {
+					CamelEditorUIActivator.pluginLog().logError(e);
+				}
+            }
+        }
+		return null;
+	}
+
+    public static Parameter createParameter(String name, String jType) {
+    	Parameter outParm = new Parameter();
+    	outParm.setName(name);
+    	outParm.setJavaType(jType);
+    	return outParm;
+    }
+
+    public static boolean hasMethod(String methodName, IType type) {
+		if (type == null) {
+			return false;
+		}
+
+		SearchPattern pattern = SearchPattern.createPattern(
+				methodName, 
+				IJavaSearchConstants.METHOD, 
+				IJavaSearchConstants.DECLARATIONS, 
+				SearchPattern.R_EXACT_MATCH);
+
+		IJavaSearchScope scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { type });
+
+		CountingSearchRequestor matchCounter = new CountingSearchRequestor();
+
+		SearchEngine search = new SearchEngine();
+		try {
+			search.search(pattern, 
+					new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() }, 
+					scope, 
+					matchCounter, 
+					null);
+		} catch (CoreException ce) {
```

Comment:
effectively, looks like a bug in Sonar/github comment that it is not removing it after fixing it